packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts

```diff
@@ -0,0 +1,355 @@
+    };
+
+    const updateUnreferencedDataStores = (id: string) => {
+        if (!unrefDataStores.has(id) && !hasReferencedParent(id)) {
+            unrefDataStores.add(id);
+            refDataStores.delete(id);
+            const testNode = getTestNode(id);
+            for (const childId of testNode.children) {
+                updateUnreferencedDataStores(childId);
+            }
+        }
+    };
+
+    const referenceDataStore = (dataStore: TestDataObject, parentDataStore: TestDataObject) => {
+        parentDataStore._root.set(dataStore.id, dataStore.handle);
+        getTestNode(parentDataStore.id).children.add(dataStore.id);
+        getTestNode(dataStore.id).parents.add(parentDataStore.id);
+        if (refDataStores.has(parentDataStore.id)) {
+            updateReferencedDataStores(dataStore.id);
+        }
+    };
+
+    const unreferenceDataStore = (dataStore: TestDataObject, parentDataStore: TestDataObject) => {
+        if (rootDataStores.has(dataStore.id)) {
+            return;
+        }
+        parentDataStore._root.delete(dataStore.id);
+        getTestNode(parentDataStore.id).children.delete(dataStore.id);
+        getTestNode(dataStore.id).parents.delete(parentDataStore.id);
+        updateUnreferencedDataStores(dataStore.id);
+    };
+
+    const getRandomContainer = (): IContainer => {
+        const id = randomInteger(0, connectedContainers.length - 1);
+        return connectedContainers[id];
+    };
+
+    const getRandomDataStoreFromContainer = async (container: IContainer) => {
+        const random = randomInteger(0, testNodes.size - 1);
+        const dataStoreId = Array.from(testNodes.values())[random].id;
+        return requestFluidObject<TestDataObject>(container, dataStoreId);
+    };
+
+    const getRandomParentForDataStore = async (dataStore: TestDataObject, container: IContainer) => {
+        const parents = getTestNode(dataStore.id).parents;
+        const random = randomInteger(0, parents.size - 1);
+        const parentId = Array.from(parents.values())[random];
+        return requestFluidObject<TestDataObject>(container, parentId);
+    };
+
+    const referenceRandomDataStore = async () => {
+        if (connectedContainers.length <= 0 || testNodes.size <= 0) {
+            return;
+        }
+        const container = getRandomContainer();
+        const dataStore = await getRandomDataStoreFromContainer(container);
+        const parent = await getRandomDataStoreFromContainer(container);
+        referenceDataStore(dataStore, parent);
+    };
+
+    const unreferenceRandomDataStore = async () => {
+        if (connectedContainers.length <= 0 || testNodes.size <= 0) {
+            return;
+        }
+        const container = getRandomContainer();
+        const dataStore = await getRandomDataStoreFromContainer(container);
+        const parentDataStore = await getRandomParentForDataStore(dataStore, container);
+        unreferenceDataStore(dataStore, parentDataStore);
+    };
```

Comment:
This could be a no-op - maybe plumb through a boolean to indicate if it successfully removed a link?  e.g. if it selects a root or already-unreferenced `dataStore` (one that has no parents) then the caller might want to know.  Or alternatively this could loop until it removes a link - but then it could get in an infinite loop if there are no links!  Something to think through.